client/state/comments/selectors.js

```diff
@@ -108,16 +120,32 @@ export const getPostCommentsTree = createSelector(
 				? filter( allItems, item => item.isPlaceholder || item.status === status )
 				: allItems;
 
-		return {
-			...keyBy(
-				map( items, item => ( {
```

Comment:
a `map(items)` inside of a `map( items )` eeek!

Reply:
While the new code is definitely much faster, I don't think we needed to make such a drastic change from the declarative approach.

It could be possible to extract the inner loop. FFTI since I'll be out the next few days, but heres my idea in pseudocode:

```js
const commentsByParent = keyBy( items, 'parent.ID' ); // this is the equivalent of the inner loop
return ...keyBy(
  map( items, comment => ({
    children: get( commentsByParent, comment.ID, [] ),
    data: item,
  })
)
```

Reply:
reworked in https://github.com/Automattic/wp-calypso/pull/18303/commits/cbb4f7ecf7e290348e18be3aa1e09f00f464b07c